src/core/TChem_Driver.cpp

```diff
@@ -0,0 +1,279 @@
+  g_tchem->freeGasKineticModel();
+}
+
+void TChem::Driver::freeGasKineticModel() {
+  _chem_file = std::string();
+  _therm_file = std::string();
+}
+
+void TChem::Driver::createStateVector() {
+  const ordinal_type len = TChem::Impl::getStateVectorSize(_kmcd_host.nSpec);
+  const ordinal_type nBatch = 1;
+  _state = real_type_2d_view_host("state dev", nBatch, len);
+  for (ordinal_type k = 0; k < len; k++) {
+    _state(0,k) = 0.0;
+  }
+}
+
+/* Get the state vector */
+auto TChem::Driver::getStateVector() {
+  auto state_at_0 = Kokkos::subview(_state, 0, Kokkos::ALL);
+  return state_at_0;
+}
+
+void TChem_getStateVector(TChem::real_type *state){
+  auto q = g_tchem->getStateVector();
+  auto len = TChem_getLengthOfStateVector();
+  for (ordinal_type k = 0; k < len; k++) {
+     state[k] = q[k];
+  }
+}
+
+/* Set the values of the state vector */
+void TChem_setStateVector(double *array){
+  g_tchem->setStateVector(array);
+}
+
+void TChem::Driver::setStateVector(double *array) {
+  auto len = TChem_getLengthOfStateVector();
+  for (ordinal_type k = 0; k < len; k++){
+     _state(0,k) = array[k];
+  }
+}
+
+/* Return species name */
+int TChem_getSpeciesName(int * index, char* result, const std::string::size_type buffer_size){
+  std::string specName = g_tchem->getSpeciesName(index);  
+  specName.copy(result, buffer_size);
+  result[specName.length()] = '\0';
+  return specName.length();
+}
+
+std::string TChem::Driver::getSpeciesName(int *index){
+  const auto speciesNamesHost = Kokkos::create_mirror_view(_kmcd_host.speciesNames);
+  Kokkos::deep_copy(speciesNamesHost, _kmcd_host.speciesNames);
+  int k = *index;
+  std::string species_name = &speciesNamesHost(k,0);
+  return species_name;
+}
+
+/* Return number of species */
+ordinal_type TChem_getNumberOfSpecies(){
+   ordinal_type nSpec = g_tchem->getNumberOfSpecies();
+   return nSpec;
+}
+
+ordinal_type TChem::Driver::getNumberOfSpecies() { return _kmcd_host.nSpec;}
+
+/* Return length of the state vector */
+int TChem_getLengthOfStateVector() { 
+  return g_tchem == nullptr ? -1 : g_tchem->getLengthOfStateVector();
+}
+
+ordinal_type TChem::Driver::getLengthOfStateVector() const {
+  return Impl::getStateVectorSize(_kmcd_host.nSpec);
+}
+
+/* Integrate a time step */
+void TChem_doTimestep(const double &del_t){
+  g_tchem->doTimestep(del_t);
+}
+
+/* Integrate a time step */
+void TChem::Driver::doTimestep(const double del_t){
+
+  const auto exec_space_instance = TChem::exec_space();
+//  using device_type = typename Tines::UseThisDevice<exec_space>::type;
+  using device_type = typename Tines::UseThisDevice<TChem::exec_space>::type;
+  using interf_host_device_type = typename Tines::UseThisDevice<TChem::host_exec_space>::type;
+  using problem_type = TChem::Impl::AtmosphericChemistry_Problem<real_type, interf_host_device_type>;
+  using policy_type = typename TChem::UseThisTeamPolicy<TChem::exec_space>::type;
+  policy_type policy(exec_space_instance, 1, Kokkos::AUTO());
```

Comment:
Here, we are assuming the number of batches is equal to 1.

Reply:
`Kokkos::AUTO()` does not provide the best performance. We may need to pass a team_size variable to this routine.

Reply:
Fixed batch size in [42a7726](https://github.com/PCLAeroParams/TChem-atm/pull/55/commits/42a772624936477c979abcfe3345687edeb32e66). Also corrected other variables that had hardcoded batch size of 1.

Reply:
> `Kokkos::AUTO()` does not provide the best performance. We may need to pass a team_size variable to this routine.

Added something in [fe2c156](https://github.com/PCLAeroParams/TChem-atm/pull/55/commits/fe2c15617bc8091ac3b827b6903f21eb62b55783) for now where these are now able to be specified in the yaml file with other solver information.